pkg/detectors/twilio/twilio.go

```diff
@@ -71,46 +73,20 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			}
 
 			if verify {
-				client := s.client
-				if client == nil {
-					client = defaultClient
-				}
+				extraData, isVerified, verificationErr := verifyTwilio(ctx, s.getClient(), key, sid)
+				s1.Verified = isVerified
+				s1.SetVerificationError(verificationErr)
 
-				req, err := http.NewRequestWithContext(
-					ctx, "GET", "https://verify.twilio.com/v2/Services", nil)
-				if err != nil {
-					continue
+				for key, value := range extraData {
+					s1.ExtraData[key] = value
 				}
-				req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
-				req.Header.Add("Accept", "*/*")
-				req.SetBasicAuth(sid, key)
-				res, err := client.Do(req)
-				if err == nil {
-					defer res.Body.Close()
-
-					if res.StatusCode >= 200 && res.StatusCode < 300 {
-						s1.Verified = true
-						s1.AnalysisInfo = map[string]string{"key": key, "sid": sid}
-						var serviceResponse serviceResponse
-						if err := json.NewDecoder(res.Body).Decode(&serviceResponse); err == nil && len(serviceResponse.Services) > 0 { // no error in parsing and have at least one service
-							service := serviceResponse.Services[0]
-							s1.ExtraData["friendly_name"] = service.FriendlyName
-							s1.ExtraData["account_sid"] = service.AccountSID
-						}
-					} else if res.StatusCode == 401 || res.StatusCode == 403 {
-						// The secret is determinately not verified (nothing to do)
-					} else {
-						err = fmt.Errorf("unexpected HTTP response status %d", res.StatusCode)
-						s1.SetVerificationError(err, key)
-					}
-				} else {
-					s1.SetVerificationError(err, key)
+
+				if s1.Verified {
+					s1.AnalysisInfo = map[string]string{"key": key, "sid": sid}
 				}
 			}
 
-			if len(keyMatches) > 0 {
```

Comment:
I am not sure why this check was added initially as if we are inside `keyMatches` loop it will always be true